src/utils/promise.js

```diff
@@ -51,7 +51,7 @@ export class GreedyPromise {
       const continuation = () => {
         let value = result[1];
         let [handler, resolveFn] = result[0] === SUCCESS ? [onSuccess, resolve] : [onError, reject];
-        if (typeof handler === 'function') {
+        if (typeof handler === 'function' && value !== undefined && value !== null) {
```

Comment:
Hrmmmmmm... well do we know why in our case on the first call this value could be undefined or null?